src/form/components/Textarea/index.js

```diff
@@ -1,131 +1,99 @@
-      styles.flexBasis = `calc(100% - ${labelwidth} - 20px)`;
-    }
-
-    return styles;
-  }
-
-  getTextareaError() {
-    const { error } = this.props;
+  const getNonLabelWidth = () => {
+    if (labelwidth) return { flexBasis: `calc(100% - ${labelwidth} - 20px)` };
+    return {};
+  };
 
+  function getTextareaError() {
     return (error && error !== 'true' && error !== 'false')
       ? (
-        <p
-          className="tyk-form-control__error-message"
-        >
+        <p className="tyk-form-control__error-message">
           { error }
         </p>
       )
       : null;
   }
 
-  render() {
-    const {
-      id,
-      input,
-      label,
-      note,
-      value,
-      readOnly,
-      ...rest
-    } = this.props;
-
-    return (
-      <Fragment>
-        <div className={this.getCssClasses()}>
+  return (
+    <div className={classes}>
+      {
+        label
+          ? <label htmlFor={id} style={getLabelStyles()}>{ label }</label>
+          : null
+      }
+      {!readOnly && (
+        <div
+          className="tyk-form-control__wrapper"
+          style={getNonLabelWidth()}
+        >
+          <textarea
+            className="tyk-form-control"
+            {...rest}
+            {...input}
+          >
+            { value }
+          </textarea>
           {
-            label
-              ? <label htmlFor={id} style={this.getLabelStyles()}>{ label }</label>
+            note
+              ? <p className="tyk-form-control__help-block">{ note }</p>
               : null
           }
-          {!readOnly && (
-            <div
-              className="tyk-form-control__wrapper"
-              style={this.getNonLabelWidth()}
-            >
-              <textarea
-                className="tyk-form-control"
-                {...rest}
-                {...input}
-              >
-                { value }
-              </textarea>
-              {
-                note
-                  ? <p className="tyk-form-control__help-block">{ note }</p>
-                  : null
-              }
-              {this.getTextareaError()}
-            </div>
-          )}
-          {readOnly && <div className="tyk-form-control--readonly">{value || '-'}</div>}
+          {getTextareaError()}
         </div>
-      </Fragment>
-    );
-  }
+      )}
+      {readOnly && <div className="tyk-form-control--readonly">{value || '-'}</div>}
+    </div>
+  );
 }
+
+Textarea.propTypes = {
+  disabled: PropTypes.bool,
+  readOnly: PropTypes.bool,
+  input: PropTypes.instanceOf(Object),
+  error: PropTypes.string,
+  label: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+    PropTypes.element,
+    PropTypes.func,
+    PropTypes.string,
+  ]),
+  labelwidth: PropTypes.string,
+  name: PropTypes.string,
+  note: PropTypes.string,
+  onChange: PropTypes.func,
+  id: PropTypes.string,
+  theme: PropTypes.string,
+  value: PropTypes.string,
+  wrapperClassName: PropTypes.string,
+};
```

Comment:
not a biggie, but we can add comments here to be picked up by styleguidist doc

Reply:
will do, thanks